zilliqa/src/consensus.rs

```diff
@@ -2048,10 +2173,14 @@ impl Consensus {
     // TODO: Consider if these checking functions should be implemented at the deposit contract level instead?
 
     fn check_quorum_in_bits(
-        &self,
+        &mut self,
```

Comment:
You can do this without the `&mut self` by instead getting the state with `let state = self.state.at_root(parent_state_hash.into());`

Maybe there are other places this pattern can be applied too?

Reply:
Right. I've seen both patterns used in the code. I'll make this change tomorrow.